packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -678,6 +678,8 @@ bool validateAgpAndKgp(Logger logger, {required String? kgpV, required String? a
   // Continuous KGP version handling is prefered in case an emergency patch to a
   // past release is shipped this code will assume the version range that is closest.
 
+  // When documenting a new KGP and AGP compatibility max, be sure to update the
+  // maxKnownAgpVersionWithFullKotlinSupport with the new AGP max.
```

Comment:
This comment is helpful but what would be better would be a test that compares `templateAndroidGradlePluginVersion` with `maxKnownAgpVersionWithFullKotlinSupport` and makes sure that max is greater than template. 

While you are at it please add a test for `maxKnownAndSupportedAgpVersion` is equal to or less than `maxKnownAgpVersion` and a test (or one test that verifies all conditions) that ensures that `maxKnownAgpVersionWithFullKotlinSupport` is less than `maxKnownAndSupportedAgpVersion`.